server/service/src/sync/translations/master_list.rs

```diff
@@ -31,16 +70,47 @@ impl SyncTranslation for MasterListTranslation {
 
         let data = serde_json::from_str::<LegacyListMasterRow>(&sync_record.data)?;
 
-        let result = MasterListRow {
-            id: data.ID,
-            name: data.description,
-            code: data.code,
-            description: data.note,
+        let master_list = MasterListRow {
+            id: data.id.clone(),
+            name: data.description.clone(),
+            code: data.code.clone(),
+            description: data.note.clone(),
         };
 
-        Ok(Some(IntegrationRecords::from_upsert(
-            PullUpsertRecord::MasterList(result),
-        )))
+        let generate = generate_requisition_program(connection, &data)?;
+        let (program, program_requisition_settings, program_requisition_order_types) =
+            match generate {
+                Some(generate) => (
+                    generate.program_row,
+                    generate.program_requisition_settings_row,
+                    generate.program_requisition_order_type_rows,
+                ),
+                None => {
+                    return Ok(Some(IntegrationRecords::from_upsert(
```

Comment:
nit:
If we need to create any other records from a master list in future, it might be easy to miss that this return.
I wonder if it's better to create the upserts Vec with `PullUpsertRecord::MasterList(master_list)` and then push the other records in the match statement?

That would be more intuitive to be *I think*

Reply:
That being said I love a good early return :)

Normally early returns are better for situations where we're handling an error and don't want to continue though IMHO